code/main/edge/edgeModule.brs

```diff
@@ -28,22 +28,33 @@ function _adb_EdgeModule(configurationModule as object, identityModule as object
         return invalid
     end if
 
+    if not _adb_isConsentStateModule(consentState) then
+        _adb_logError("EdgeModule::_adb_EdgeModule() - consentState is not valid.")
+        return invalid
+    end if
+
     edgeResponseManager = _adb_EdgeResponseManager()
 
     module = _adb_AdobeObject("com.adobe.module.edge")
     module.Append({
         _EDGE_REQUEST_PATH: "/v1/interact",
+        _REQUEST_TYPE_EDGE: "edge",
         _configurationModule: configurationModule,
         _identityModule: identityModule,
         _edgeResponseManager: edgeResponseManager,
-        _edgeRequestWorker: _adb_EdgeRequestWorker(edgeResponseManager),
+        _edgeRequestWorker: _adb_EdgeRequestWorker(edgeResponseManager, consentState),
 
         ' sendEvent API triggers this API to queue edge requests
         ' requestId: unique id for the request
         ' eventData: data to be sent to edge
         ' timestampInMillis: timestamp of the event
-        processEvent: function(requestId as string, eventData as object, timestampInMillis as longinteger) as void
-            m._edgeRequestWorker.queue(requestId, eventData, timestampInMillis, {}, m._EDGE_REQUEST_PATH)
+        processEvent: function(requestId as string, eventData as object, timestampInMillis as longinteger, requestType = m._REQUEST_TYPE_EDGE as string) as void
+            edgeRequest = _adb_EdgeRequest(requestId, eventData, timestampInMillis)
+            edgeRequest.setMeta({})
```

Comment:
nit: Is this necessary? Can't you handle it when processing the request?

Reply:
yes, we do update it while processing